ReindexIT#testDeleteByQuery and ReindexIT#testUpdateByQuery failures

**Build scan**:

https://gradle-enterprise.elastic.co/s/jbzekcqwjf7xc
https://gradle-enterprise.elastic.co/s/kbwxhs4vp764k
https://gradle-enterprise.elastic.co/s/cxcldr6dfkzba
https://gradle-enterprise.elastic.co/s/wwzzzkafjmkb4

**Repro line**:

e.g. `REPRODUCE WITH: ./gradlew ':client:rest-high-level:asyncIntegTestRunner' --tests "org.elasticsearch.client.ReindexIT.testDeleteByQuery" -Dtests.seed=E520D301C9568B27 -Dtests.security.manager=true -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=ru -Dtests.timezone=EST5EDT -Druntime.java=8`

**Reproduces locally?**:

No. There's nothing particularly helpful in the node logs either AFAICT.

**Applicable branches**:

Only seen on 7.8 and 7.9.

**Failure history**:

[Four failures](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-90d,mode:quick,to:now))&_a=(columns:!(test.failed-testcases,build.branch),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'(testDeleteByQuery%20OR%20testUpdateByQuery)%20AND%20%22a%20collection%20with%20size%22'),sort:!(process.time-start,desc))) over the last 90 days. Maybe duplicates #46301?

**Failure excerpt**:

```
org.elasticsearch.client.ReindexIT > testDeleteByQuery FAILED |  
-- | --
  | java.lang.AssertionError: |  
  | Expected: a collection with size <1> |  
  | but: collection size was <0> |  
  | at __randomizedtesting.SeedInfo.seed([E520D301C9568B27:EBEFD62A9099F4B1]:0) |  
  | at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
  | at org.junit.Assert.assertThat(Assert.java:956) |  
  | at org.junit.Assert.assertThat(Assert.java:923) |  
  | at org.elasticsearch.client.ReindexIT.testDeleteByQuery(ReindexIT.java:421)

```

The failure indicates that the rethrottle request did not return the expected task. However we know it's running before the rethrottle request since we call `findTaskToRethrottle`, and we also know it's running afterwards since subsequent tests report `There are still tasks running after this test that might break subsequent tests [indices:data/write/delete/byquery].`

--- a/client/rest-high-level/src/test/java/org/elasticsearch/client/ReindexIT.java
+++ b/client/rest-high-level/src/test/java/org/elasticsearch/client/ReindexIT.java
@@ -50,6 +50,7 @@ import org.elasticsearch.tasks.TaskId;
 
 import java.io.IOException;
 import java.util.Collections;
+import java.util.List;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
@@ -283,7 +284,7 @@ public class ReindexIT extends ESRestHighLevelClientTestCase {
                 }
             });
 
-            TaskId taskIdToRethrottle = findTaskToRethrottle(UpdateByQueryAction.NAME);
+            TaskId taskIdToRethrottle = findTaskToRethrottle(UpdateByQueryAction.NAME, updateByQueryRequest.getDescription());
             float requestsPerSecond = 1000f;
             ListTasksResponse response = execute(new RethrottleRequest(taskIdToRethrottle, requestsPerSecond),
                 highLevelClient()::updateByQueryRethrottle, highLevelClient()::updateByQueryRethrottleAsync);
@@ -414,7 +415,7 @@ public class ReindexIT extends ESRestHighLevelClientTestCase {
                 }
             });
 
-            TaskId taskIdToRethrottle = findTaskToRethrottle(DeleteByQueryAction.NAME);
+            TaskId taskIdToRethrottle = findTaskToRethrottle(DeleteByQueryAction.NAME, deleteByQueryRequest.getDescription());
             float requestsPerSecond = 1000f;
             ListTasksResponse response = execute(new RethrottleRequest(taskIdToRethrottle, requestsPerSecond),
                 highLevelClient()::deleteByQueryRethrottle, highLevelClient()::deleteByQueryRethrottleAsync);
@@ -477,7 +478,7 @@ public class ReindexIT extends ESRestHighLevelClientTestCase {
         }
     }
 
-    private static TaskId findTaskToRethrottle(String actionName) throws IOException {
+    private static TaskId findTaskToRethrottle(String actionName, String description) throws IOException {
         long start = System.nanoTime();
         ListTasksRequest request = new ListTasksRequest();
         request.setActions(actionName);
@@ -485,13 +486,16 @@ public class ReindexIT extends ESRestHighLevelClientTestCase {
         do {
             ListTasksResponse list = highLevelClient().tasks().list(request, RequestOptions.DEFAULT);
             list.rethrowFailures("Finding tasks to rethrottle");
+            List<TaskGroup> taskGroups =
+                list.getTaskGroups().stream()
+                    .filter(taskGroup -> taskGroup.getTaskInfo().getDescription().equals(description)).collect(Collectors.toList());
             assertThat("tasks are left over from the last execution of this test",
-                list.getTaskGroups(), hasSize(lessThan(2)));
-            if (0 == list.getTaskGroups().size()) {
+                taskGroups, hasSize(lessThan(2)));
+            if (0 == taskGroups.size()) {
                 // The parent task hasn't started yet
                 continue;
             }
-            TaskGroup taskGroup = list.getTaskGroups().get(0);
+            TaskGroup taskGroup = taskGroups.get(0);
             assertThat(taskGroup.getChildTasks(), empty());
             return taskGroup.getTaskInfo().getTaskId();
         } while (System.nanoTime() - start < TimeUnit.SECONDS.toNanos(10));

